aisdc/attacks/likelihood_attack.py

```diff
@@ -296,6 +320,19 @@ def run_scenario_from_preds(  # pylint: disable = too-many-statements, too-many-
                         # catch-all
                         shadow_row_to_confidence[i].append(_logit(0))
 
+            lengths_shadow_row_to_confidence = {
```

Comment:
this is the bulk of the failfast code and it's hard to follow without comments -- could you add some? Also missing some overall documentation (in the codebase) about how this failfast works (I think I missed the meeting).

I _think_ what's happening is that you stop when you have at least a mininum number of confidence values for each example (which may save some time due to the random nature of the dataset splitting) but I'm not sure -- needs clarification so that a user will know when to use this.

Reply:
Added some comments to explain the process.